Source/WebCore/platform/graphics/gstreamer/mse/MediaSourcePrivateGStreamer.cpp

```diff
@@ -136,19 +136,35 @@ void MediaSourcePrivateGStreamer::markEndOfStream(EndOfStreamStatus endOfStreamS
     }
     GST_DEBUG_OBJECT(m_playerPrivate.pipeline(), "Marking EOS, status is %s", statusString);
 #endif
-    if (endOfStreamStatus == EosNoError)
+    if (endOfStreamStatus == EosNoError) {
         m_playerPrivate.setNetworkState(MediaPlayer::NetworkState::Loaded);
+
+        auto bufferedRanges = buffered();
+        if (!bufferedRanges.length()) {
+            GST_DEBUG("EOS with no buffers");
+            m_playerPrivate.setEosWithNoBuffers(true);
+        }
+    }
     m_isEnded = true;
+    MediaSourcePrivate::markEndOfStream(endOfStreamStatus);
 }
 
 void MediaSourcePrivateGStreamer::unmarkEndOfStream()
 {
     ASSERT(isMainThread());
+<<<<<<< HEAD
```

Comment:
Looks like a merge conflict in the code

Reply:
This was my fault. To correct it, I force-pushed wpe-2.38 to the previous commit to the one with the merge conflict, and then pushed the good commit on top, as if the merge conflict had never happened.
Maybe this PR needs to be rebased and resubmitted.

Reply:
OK, I've rebased my branch